apps/sms/views/shared/js/contacts.js

```diff
@@ -116,7 +116,7 @@
          *
          */
         return Promise.resolve(
-          filter.filterValue === undefined ? null : []
+          filter.filterValue === undefined ? [] : []
         );
       }
 
```

Comment:
The easiest part of this bug is to change Contacts, the next step is to skim through entire Messages app and find places where we use ```Contacts``` methods and verify results. The main purpose of this bug is to simplify all those places, I'd say there are plenty of them :)

So, for example see [1] and [2].

In [1] you need to change ```!contacts || !contacts.length``` to just ```!contacts.length```. In [2] the change will be ```var isContact = contacts !== null && !!contacts.length``` to just ```var isContact = contacts.length > 0```

Also don't forget to check that all tests are passing, Messages app has [integration and unit tests](https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/Testing_Gaia_code_changes). Every time you update PR you'll see comment from ```lightsofapollo``` that looks like this:
```
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=119c7266d0b009547a3f7c9f653cd31f35312ea5
```

That gives you link [3] that you can use to verify that all tests are green (see more info on [Treeherder tool](https://developer.mozilla.org/en-US/Firefox_OS/Automated_testing/Treeherder)), so for example I see that at least 3 related jobs are failing: <b>GU21</b> (gaia unit test suite 21), <b>Gij11</b> (gaia integration javascript tests) and <b>JSHINT</b>. You can tap on any of this jobs to see more details.

[1] https://github.com/mozilla-b2g/gaia/blob/8a1e4ae522c121c5cacd39b20a5386ec9055db82/apps/sms/views/conversation/js/conversation.js#L669
[2] https://github.com/mozilla-b2g/gaia/blob/8a1e4ae522c121c5cacd39b20a5386ec9055db82/apps/sms/views/conversation/js/information.js#L453
[3] https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=119c7266d0b009547a3f7c9f653cd31f35312ea5